xds/src/main/java/com/linecorp/armeria/xds/ResourceParser.java

```diff
@@ -28,17 +28,17 @@
 
 import com.linecorp.armeria.common.annotation.Nullable;
 
-abstract class ResourceParser {
+abstract class ResourceParser<T extends ResourceHolder<?>> {
```

Comment:
Well the T (`ResourceHolder`) is the same type that `XdsStreamSubscriber` uses.
https://github.com/line/armeria/blob/012b9d14c9d6d6ef7004b60c4adbbf8652455e36/xds/src/main/java/com/linecorp/armeria/xds/DefaultResponseHandler.java#L53-L59

 Let me try adding one more type to `ResourceParser` then.

Reply:
Ah... Let me check first before adding another type.

Reply:
Should we change the type of `onData()` instead?
```
class XdsStreamSubscriber<T extends Message> implements SafeCloseable {

    void onData(ResourceHolder<T> data) {
    }
}
```

Reply:
Ditto I'm not sure if it's a good idea to use `ResourceHolder<T>` instead of its subclasses because some of the API has its own methods. e.g. (`ListenerResourceHolder.connectionManager()`, ClusterResourceHolder.upstreamTlsContext())
We should cast them once more.